doc/specs/#3327 - Application Theming/#3327 - Application Theming.md

```diff
@@ -0,0 +1,500 @@
+---
+author: Mike Griese @zadjii-msft
+created on: 2019-12-13
+last updated: 2020-04-23
+issue id: #3327
+---
+
+# Spec Title
+
+## Abstract
+
+This spec outlines how the Windows Terminal will enable users to create custom
+"themes" for the application, enabling further customization of the window.
+These themes will be implemented as objects containing a group of UI-specific
+properties, so users can quickly apply a group of properties atomically.
+
+## Inspiration
+
+Much of the inspiration for this feature comes from VsCode and its themes. These
+themes can be more than just different color palettes for the editor - these
+themes can control the appearance of a variety of UI elements of the VsCode
+window.
+
+
+## Solution Design
+
+### Requested Functionality ("User Stories")
+
+The following is a long list of ideas of elements of the window that the user
+should be able to customize:
+
+* [ ] Pane Border colors (both the background, and the "focused" color) ([#3061])
+* [ ] Pane border width ([#3062])
+* [ ] Tab _Row_ and _Item_ Background color ([#702]/[#1337]/[#2994]/[#3774]/[#1963])
+    - Some users want to set these to the accent color
+    - Some users want to set these to a specific custom color
+    - Some users want this to use the color straight from the active Terminal,
+      allowing the tab or titlebar to "blend into" the terminal
+* [ ] Feature Request: Setting to hide/remove close ("x") button from tabs ([#3335])
+* [ ] Various different tab sizing modes
+    - the current sizing, which is `SizeToContent`
+    - Setting a min/max width on tabs
+    - Configuring tabs to split the available space
+
+Other lower-priority ideas:
+* [ ] Enable hiding the tab icon altogether
+* [ ] Enable forcing tab icons to monochrome
+* [ ] Tab row height

```

Comment:
🦙 The Tab width behavior would fit in here as well (Compact, Equal, SizeToContent).

Reply:
Oh, that's a good point. We'll need to come up with a way of handling what happens when the user has it set in both the `theme` and directly in the global settings - probably makes sense to just use the global value to override the theme value, right?

Reply:
@zadjii-msft or does it make sense to just move the setting itself into the theme (and have some default value if the theme doesn't provide one)? Or use the 'global' setting as the fallback value if the theme doesn't contain one?